network/oauth_client.go

```diff
@@ -64,6 +64,7 @@ func (oc OAuthClient) Do(request *http.Request) (*http.Response, error) {
 	}
 
 	client := oc.oauthConfig.Client(oc.context, token)
+	client.Timeout =  oc.timeout
```

Comment:
No sir, it actually is added. `oauth2` implements its own transport - slightly confusing:

https://github.com/golang/oauth2/blob/master/internal/transport.go#L35-L39